src/DataGrid.tsx

```diff
@@ -141,7 +141,7 @@ export interface DataGridProps<R, SR = unknown, K extends Key = Key> extends Sha
   rowGrouper?: ((rows: readonly R[], columnKey: string) => Record<string, readonly R[]>) | null;
   expandedGroupIds?: ReadonlySet<unknown> | null;
   onExpandedGroupIdsChange?: ((expandedGroupIds: Set<unknown>) => void) | null;
-  onFill?: ((event: FillEvent<R>) => R[]) | null;
+  onFill?: ((event: FillEvent<R>) => R) | null;
   onPaste?: ((event: PasteEvent<R>) => R) | null;
```

Comment:
I wonder if we should just merge them into one `onPaste` handler.
Can you think of any cases where pasing and filling would behave differently?

Reply:
For paste, columns may not be compatible so we need source and target column keys and this API may be awkward for `onFill`. Also, there won't be a way to disable them individually .